### PR TITLE
Make test fails due to keeping github.com autogenerated folder in root

### LINKED ISSUE #1707: Make test fails due to keeping github.com autogenerated folder in root
Please, answer some short questions which should help us to understand your problem / question better?

- **Which image of the operator are you using?** e.g. registry.opensource.zalan.do/acid/postgres-operator:v1.7.1
- **Where do you run it - cloud or metal? Kubernetes or OpenShift?** [Kind on Fedora Linux]
- **Are you running Postgres Operator in production?** [no]
- **Type of issue?** [Bug report]

Not sure, that's Fedora problem, but seems to `github.com` folder keeping after running `hack/update-codegen.sh` script, but need to be cleaned up or scrips need to be fixed a bit
```
09:59 $ uname -a
Linux fedora 5.14.17-301.fc35.x86_64 #1 SMP Mon Nov 8 13:57:43 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

09:42 $ make deps mocks test
GO111MODULE=on go get k8s.io/client-go@kubernetes-1.22.3
GO111MODULE=on go get github.com/golang/mock/mockgen@v1.6.0
go get: installing executables with 'go get' in module mode is deprecated.
	To adjust and download dependencies of the current module, use 'go get -d'.
	To install using requirements of the current module, use 'go install'.
	To install ignoring the current module, use 'go install' with a version,
	like 'go install example.com/cmd@latest'.
	For more information, see https://golang.org/doc/go-get-install-deprecation
	or run 'go help get' or 'go help install'.
GO111MODULE=on go mod tidy
GO111MODULE=on go mod vendor
GO111MODULE=on go generate ./...
hack/verify-codegen.sh
Generating deepcopy funcs
Generating clientset for acid.zalan.do:v1 at github.com/zalando/postgres-operator/pkg/generated/clientset
Generating listers for acid.zalan.do:v1 at github.com/zalando/postgres-operator/pkg/generated/listers
Generating informers for acid.zalan.do:v1 at github.com/zalando/postgres-operator/pkg/generated/informers
diffing hack/../pkg against freshly generated codegen
hack/../pkg up to date.
GO111MODULE=on go test ./...
# github.com/zalando/postgres-operator/github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:38:11: undefined: AWSGCPConfiguration
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:44:11: undefined: AWSGCPConfiguration
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:54:11: undefined: AdditionalVolume
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:66:11: undefined: AdditionalVolume
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:76:11: undefined: CloneDescription
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:87:11: undefined: CloneDescription
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:97:11: undefined: ConnectionPooler
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:114:11: undefined: ConnectionPooler
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:124:11: undefined: ConnectionPoolerConfiguration
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:140:11: undefined: ConnectionPoolerConfiguration
github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:140:11: too many errors
?   	github.com/zalando/postgres-operator/cmd	[no test files]
ok  	github.com/zalando/postgres-operator/pkg/apis/acid.zalan.do/v1	(cached)
ok  	github.com/zalando/postgres-operator/pkg/apiserver	(cached)
ok  	github.com/zalando/postgres-operator/pkg/cluster	(cached)
ok  	github.com/zalando/postgres-operator/pkg/controller	(cached)
ok  	github.com/zalando/postgres-operator/pkg/spec	(cached)
ok  	github.com/zalando/postgres-operator/pkg/teams	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util/config	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util/k8sutil	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util/patroni	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util/retryutil	(cached)
ok  	github.com/zalando/postgres-operator/pkg/util/teams	(cached)
make: *** [Makefile:104: test] Error 2
```

### PR DESCRIPTION
**Pull request by dmvolod**
fixes #1707

Also codegeneration scripts a bit refactored. Added `make codegen` target. Now `Run unit tests` in GitHub actions utilizing `make test` target.

### CODE REVIEW

File: pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
```
Comment (FxKu) [inline pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: I think, this line can be removed. When I run `make codegen` in this branch, I don't get this line.

**Comment by FxKu** [comment]
Awesome @dmvolod ! Thanks a lot :smiley:

File: pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go (hunk 1 above)
Comment (dmvolod) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: @FxKu seem to this line depends of the golang version utilizing. Before adding this line, CI job has failed without it.

Comment (dmvolod) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: Also saw these changes in several PRs

Comment (FxKu) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: Which CI job? Do have links to "other PRs"?

Comment (dmvolod) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: https://github.com/zalando/postgres-operator/pull/1672/files#diff-ad834b7f56540d53b0c98651ad608508b7532e395e472f12f76d5970c9ae6da7

Comment (FxKu) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go]: I see, hmm:
```
diffing hack/../pkg against freshly generated codegen
diff -Naupr hack/../pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go hack/../_tmp/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go
--- hack/../pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go	2021-12-02 07:21:30.048366700 +0000
+++ hack/../_tmp/pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go	2021-12-02 07:24:23.029626056 +0000
@@ -1,3 +1,4 @@
+//go:build !ignore_autogenerated
 // +build !ignore_autogenerated
 
 /*
hack/../pkg is out of date. Please run 'make codegen'
make: *** [Makefile:103: test] Error 1
Error: Process completed with exit code 2.
```

Comment (FxKu) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go]: on my machine it's the opposite. When I add this line `verify-codegen` fails. Maybe should update to Go 1.17.

**Comment by FxKu** [comment]
:+1:

File: pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go (hunk 1 above)
Comment (dmvolod) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go]: Yes, indeed, but confusing a bit with CI is utilizing golang version 1.16.9

Comment (FxKu) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go]: Can you bring back the generated code line we talked about? Then this PR can be merged.

Comment (dmvolod) [reply pkg/apis/acid.zalan.do/v1/zz_generated.deepcopy.go:1]: Yes, sure. Done.

**Comment by FxKu** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

**Pull request merged**